### PR TITLE
chore: refactor email tests

### PR DESCRIPTION
**Pull request by lukaisailovic**
Fix email tests and expand email testing suite

### CODE REVIEW

**Review by svenvoskamp — APPROVED** [review]
Nice! LGTM ✨

File: apps/laboratory/tests/shared/utils/email.ts [hunk 1]
```diff
@@ -1,67 +1,87 @@
-import { Mailsac, type EmailMessage } from '@mailsac/api'
+import { Mailsac } from '@mailsac/api'
+const EMAIL_CHECK_TIMEOUT = 1000
+const MAX_EMAIL_CHECK = 16
```
Comment (bkrem) [inline apps/laboratory/tests/shared/utils/email.ts:3]: Is this limit based on something in particular or just something that works?

**Review by bkrem — APPROVED** [review]
💯

File: apps/laboratory/tests/shared/utils/email.ts (hunk 1 above)
Comment (lukaisailovic) [reply apps/laboratory/tests/shared/utils/email.ts:3]: Ideally we want to keep that as low as possible to reduce the time it takes to run the tests. 10 was the initial limit, and slowly increased until I found one where almost every test would pass.

**Pull request merged**